File: Examples/test-suite/octave/li_cdata_bytes_runme.m
li_cdata_bytes

m = li_cdata_bytes.predefStr();
s = li_cdata_bytes.cdata(m, 0x200);
for i = 0:255
  if (double(s(i + 1)) != i)
    error(strcat("Value ", string(double(s(i + 1))), " should be ", string(i), " at index ", string(i)));
  end
  if (double(s(i + 1 + 256)) != i)
    error(strcat("Value ", string(double(s(i + 1 + 256))), " should be ", string(i), " at index ", string(i + 256)));
  end
end
s2 = "";
for i = 1:256
  d = char(256 - i);
  s2 = [s2 d];
end
s = [s2 s2];
m2 = li_cdata_bytes.malloc(0x200);
li_cdata_bytes.memmove(m2, s);
if (li_cdata_bytes.verifyBytes(m2) < 0)
  error("verifyBytes fail");
end
li_cdata_bytes.free(m2);

File: Examples/test-suite/kotlin/li_cdata_bytes_runme.kt
@file:JvmName("li_cdata_bytes_runme")

import li_cdata_bytes.*

fun main() {
    try {
        System.loadLibrary("li_cdata_bytes")
    } catch (e: UnsatisfiedLinkError) {
        System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e)
        kotlin.system.exitProcess(1)
    }

    val m = li_cdata_bytes.predefStr()
    val s = li_cdata_bytes.cdata(m, 0x200)
    for (i in 0 until 0x100) {
        var v = s[i].toInt() and 0xFF
        if (v != i)
            throw RuntimeException("Wrong value " + v + " at index " + i)
        v = s[i + 0x100].toInt() and 0xFF
        if (v != i)
            throw RuntimeException("Wrong value " + v + " at index " + (i + 0x100))
    }
    for (i in 0 until 0x100) {
        val v = (0x100 - 1 - i).toByte()
        s[i] = v
        s[i + 0x100] = v
    }
    val m2 = li_cdata_bytes.malloc(0x200)
    li_cdata_bytes.memmove(m2, s)
    if (li_cdata_bytes.verifyBytes(m2) != 0)
        throw RuntimeException("verifyBytes fails")
    li_cdata_bytes.free(m2)
}
